src/vs/workbench/services/textfile/browser/textFileService.ts

```diff
@@ -117,29 +114,8 @@ export abstract class TextFileService implements ITextFileService {
 	}
 
 	private beforeShutdown(): boolean | TPromise<boolean> {
-		// If hot exit is enabled then save the dirty files in the workspace and then exit
-		// Hot exit is currently disabled for both empty workspaces (#13733) and on Mac (#13305)
-		if (this.configuredHotExit && this.contextService.getWorkspace() && !platform.isMacintosh) {
-			// If there are no dirty files, clean up and exit
-			if (this.getDirty().length === 0) {
-				return this.cleanupBackupsBeforeShutdown();
-			}
-
-			return this.backupService.getWorkspaceBackupPaths().then(workspaceBackupPaths => {
-				// Only remove the workspace from the backup service if it's not the last one or it's not dirty
-				if (workspaceBackupPaths.length > 1) {
-					return this.confirmBeforeShutdown();
-				}
-
-				// Backup and hot exit
-				return this.backupAll().then(result => {
-					if (result.results.some(r => !r.success)) {
-						return true; // veto if some backups failed
-					}
-
-					return false; // the backup went smoothly, no veto
-				});
-			});
+		if (this.backupService.isHotExitEnabled) {
```

Comment:
Ok maybe `textFileServices` should use eventing or similar so that we can further decouple this into the backup service.